### PR TITLE
WIP: Add rply

### PR DESCRIPTION
**Pull request by drewokane**
This is a requirement for adding a hy recipe.

### CODE REVIEW

File: recipes/rply/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  number: 0
+  script: python setup.py install --single-version-externally-managed --record record.txt
+
+requirements:
+  build:
+    - python
+    - setuptools
+    - appdirs
+  run:
+    - python
```
Comment (synapticarbors) [inline recipes/rply/meta.yaml]: It appears as if `appdirs` is also a runtime dependency.

File: recipes/rply/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+test:
+  imports:
+    - rply
+
+about:
+  home: https://github.com/drewokane/rply
```
Comment (synapticarbors) [inline recipes/rply/meta.yaml]: I noticed that this is repo is a fork of https://github.com/alex/rply and it does not seem to contain any further updates. Can you explain the reason for the fork and why it should be the basis of this recipe vs the original repository? The pypi index is also owned by Alex.

File: recipes/rply/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  # See https://opensource.org/licenses/alphabetical
+  license: GNU
+  license_file: LICENSE
+  summary: 'A pure Python Lex/Yacc that works with RPython'
+  description: |
+    Welcome to RPLY! A pure python parser generator, that
+    also works with RPython. It is a more-or-less direct port
+    of David Beazley’s awesome PLY, with a new public API, and RPython support.
+  doc_url: https://rply.readthedocs.io/en/latest/
+  dev_url: https://github.com/drewokane/rply
```
Comment (synapticarbors) [inline recipes/rply/meta.yaml]: Again, why is this the `dev_url` rather than Alex Gaynor's original repo?

Comment (drewokane) [inline recipes/rply/meta.yaml]: Same as above.

**Comment by jakirkham** [comment]
Any chance you will have time to clean this up, @drewokane?

**Comment by drewokane** [comment]
Any chance this could get merged?

File: recipes/rply/bld.bat [hunk 4]
```diff
@@ -0,0 +1,8 @@
+"%PYTHON%" setup.py install
```
Comment (synapticarbors) [inline recipes/rply/bld.bat]: Could you remove the `bld.bat` and `build.sh` files in favor of something like:

https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L26

File: recipes/rply/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+{% set version = "0.7.4" %}
+{% set sha256 = "723303d6c5f05a7ee19f59531f66c8c7d41cfaef2676057369db1eb5520b378b" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.python.org/packages/f8/f4/c1fc57461dda9370e9600933162afb144429f7296fe7aa9bd8c8bc562934/{{ name }}-{{ version }}.tar.gz
```
Comment (synapticarbors) [inline recipes/rply/meta.yaml]: Use the url of the form:

https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L18

**Pull request merged**